cache/class-vary-cache.php

```diff
@@ -132,13 +173,15 @@ public static function clear_groups() {
 	 * @return WP_Error|boolean
 	 */
 	public static function set_group_for_user( $group, $value ) {
-		// TODO: make sure headers aren't already sent
-		// TODO: only send header if we added or changed things
-		// TODO: don't set the cookie if was already set on the request
+		if ( self::did_send_headers() ) {
+			return new WP_Error( 'did_send_headers', sprintf( 'Failed to set group (%s => %s) for user; cannot be called after the `send_headers` hook has fired.', $group, $value ) );
```

Comment:
This probably needs to be `trigger_error` It's not firing if you call it in the `template_redirect` action

Reply:
See https://github.com/Automattic/vip-go-mu-plugins/pull/1150#discussion_r259562607